colab/colab_utils.py

```diff
@@ -0,0 +1,1048 @@
+#@title Please run this cell first!
```

Comment:
Can you confirm that all functions defined in this file are needed by the notebook? There's quite a lot of code here, a lot of which looks pretty brittle. It'd be good to delete any that isn't actually used.

Reply:
Yes I confirm that all those functions are needed to run the Colab notebook, as I decided to reduce to the minimum the amount of code in the notebook. Some pieces of code can probably be even better optimized and refactored for a next future update. (e.g. progress bar, write shell scripts,...).